### DISCUSSION TITLE
Is esp32-s3 powerful enough to send mavlink and elrs together?

### DISCUSSION THREAD

**Discussion by BloodSakura3774** [ExpressLRS/ExpressLRS#2847] [Ideas]
Since ELRS already supports MAVLink telemetry radio mode now, I’m curious if I could use an esp32-s3 diversity rx and a diversity tx to do MAVLink radio and elrs control together. Mainly for minimizing uart use. I do know as for now it is only possible to do one of them at a time but just want to know if this is doable, and if it is, I would like to try and build it, both the elrs code and fc uart code of course.

**Comment by JyeSmith** [comment]
ELRS MAVLink already supports Mav and RC over a single uart and connection.  No need for 2 uarts or diversity/Gemini hardware.

You can flash master now to test.  Discord also has a lot of help available.
